### PR TITLE
refactor(provider/google): Rely on cached instance details instead of…

### PR DESCRIPTION
**Pull request by duftler**
… calling platform MIG.listInstances() api.

### CODE REVIEW

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleClusterProvider.groovy [hunk 1]
```diff
@@ -106,7 +108,9 @@ class GoogleClusterProvider implements ClusterProvider<GoogleCluster.View> {
       Keys.getClusterKey(account, application, name),
       RelationshipCacheFilter.include(SERVER_GROUPS.ns))
 
-    return clusterData ? clusterFromCacheData(clusterData, includeDetails) : null
+    def allClusterInstanceKeys = cacheView.filterIdentifiers(INSTANCES.ns, Keys.getInstanceKey(account, "*", "$name-*"))
```
Comment (lwander) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleClusterProvider.groovy]: If you're doing this sort of name matching, can you generate the name (including `*`) using Frigga? It'll make it easier to remove later.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleInstanceProvider.groovy [hunk 2]
```diff
@@ -120,9 +121,9 @@ class GoogleInstanceProvider implements InstanceProvider<GoogleInstance.View> {
       }
     }
 
-    def serverGroupKey = cacheData.relationships[SERVER_GROUPS.ns]?.first()
-    if (serverGroupKey) {
-      instance.serverGroup = Keys.parse(serverGroupKey).serverGroup
+    def serverGroup = GCEUtil.getLocalName(cacheData.attributes.metadata?.items?.find { it.key == "created-by" }?.value)
```
Comment (lwander) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleInstanceProvider.groovy]: Does this ever point to a user/service-account?

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleLoadBalancerProvider.groovy [hunk 3]
```diff
@@ -147,7 +147,9 @@ class GoogleLoadBalancerProvider implements LoadBalancerProvider<GoogleLoadBalan
           instances: [],
       )
 
-      def instanceNames = serverGroupCacheData.relationships[INSTANCES.ns]?.collect {
+      def serverGroupInstancePattern = Keys.getInstanceKey(loadBalancer.account, serverGroup.region, "$serverGroup.name-.*")
```
Comment (lwander) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleLoadBalancerProvider.groovy]: Same comment here about frigga

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleLoadBalancerProvider.groovy [hunk 4]
```diff
@@ -63,15 +63,17 @@ class GoogleLoadBalancerProvider implements LoadBalancerProvider<GoogleLoadBalan
       identifiers.addAll(serverGroup.relationships[LOAD_BALANCERS.ns] ?: [])
     }
 
+    def allApplicationInstanceKeys = cacheView.filterIdentifiers(INSTANCES.ns, Keys.getInstanceKey("*", "*", "$application-*"))
```
Comment (lwander) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleLoadBalancerProvider.groovy]: Frigga

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleClusterProvider.groovy [hunk 5]
```diff
@@ -84,11 +84,13 @@ class GoogleClusterProvider implements ClusterProvider<GoogleCluster.View> {
       }
     }
 
+    def allApplicationInstanceKeys = cacheView.filterIdentifiers(INSTANCES.ns, Keys.getInstanceKey("*", "*", "$applicationName-*"))
```
Comment (lwander) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleClusterProvider.groovy]: Oh boy poor moniker

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleClusterProvider.groovy [hunk 6]
```diff
@@ -151,23 +156,20 @@ class GoogleClusterProvider implements ClusterProvider<GoogleCluster.View> {
...
 
       def serverGroupData = cacheView.getAll(SERVER_GROUPS.ns, serverGroupKeys, filter)
 
       def securityGroups = securityGroupProvider.getAllByAccount(false, clusterView.accountName)
 
-      def instanceKeys = serverGroupData.collect { serverGroup ->
-        serverGroup.relationships[INSTANCES.ns] as List<String>
-      }.flatten()
-      def instances = instanceProvider.getInstances(clusterView.accountName, instanceKeys, securityGroups)
+      def clusterInstancePattern = Keys.getInstanceKey(clusterView.accountName, ".*", "$clusterView.name-.*")
```
Comment (lwander) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleClusterProvider.groovy]: sweating

**Comment by duftler** [comment]
As per offline discussion, we have decided to leave TODOs so we remember to de-frigga these new instances along with everything else, as opposed to forcing the frigga logic in.

**Comment by duftler** [comment]
This can only help moniker grow up big and strong.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleInstanceProvider.groovy (hunk 2 above)
Comment (duftler) [reply clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleInstanceProvider.groovy]: Not as far as I can tell. I have added a TODO to replace this with whatever unique key we end up adding to enable easier grouping via stackdriver (this is happening shortly).

**Comment by lwander** [comment]
m4acs99

**Pull request merged**